### PR TITLE
Possibility to use TwistStamped (still Twist by default)

### PR DESCRIPTION
**Pull request by klintan**
Ros2_control decided to go for TwistStamped (which I can understand because of the extra info + timestamp you get) however that makes it incompatible with this library. This is a suggested fix for that, if you feel this is an option, i'll go ahead and change the other modules (joy and  mouse) the same way.

Let me know what you think :) 

PS. would be good to create a foxy-devel branch as well.

### CODE REVIEW

File: key_teleop/key_teleop/key_teleop.py [hunk 1]
```diff
@@ -124,7 +125,13 @@ def __init__(self, interface):
         super().__init__('key_teleop')
 
         self._interface = interface
-        self._pub_cmd = self.create_publisher(Twist, 'key_vel')
+
+        self._twist_stamp_enabled = self.declare_parameter('twist_stamped_enabled', False).value
```
Comment (bmagyar) [inline key_teleop/key_teleop/key_teleop.py:129]:
```suggestion
        self._publish_stamped_twist = self.declare_parameter('publish_stamped_twist', False).value
```
I'd suggest renaming this variable and the parameter name for making it even clearer

File: key_teleop/key_teleop/key_teleop.py [hunk 2]
```diff
@@ -161,6 +168,17 @@ def _get_twist(self, linear, angular):
         twist.angular.z = angular
         return twist
 
+    def _get_twist_stamped(self, linear, angular):
```
Comment (bmagyar) [inline key_teleop/key_teleop/key_teleop.py:171]: Typically, `getXYZ` is supposed to fetch a value from an object. Here, you are making a new twist stamped object from the arguments, I suggest renaming
```suggestion
    def _make_twist_stamped(self, linear, angular):
```

Comment (bmagyar) [reply key_teleop/key_teleop/key_teleop.py:171]: At the same time, we may as well rename `_get_twist` while at it...

**Review by bmagyar — CHANGES_REQUESTED** [review]
The changes look good in general, please rebase to latest master & have a look at my comments.

**Comment by klintan** [comment]
> The changes look good in general, please rebase to latest master & have a look at my comments.

Sorry for the slow response on this one. It should all be addressed now as far as the feedback goes.

**Comment by Rayman** [comment]
This PR references https://github.com/ros-planning/navigation2/issues/1594. See also the discussion there.

**Review by bmagyar — APPROVED** [review]

**Comment by Rayman** [comment]
@bmagyar this PR breaks compatibility right? So you might only want this PR for galactic

**Pull request merged**